### PR TITLE
chore(back-end): LFA feedback

### PR DESCRIPTION
**Pull request by Raymond-Osy**
#### What does this PR do?
- It is for the feedback given by the LFA for my backend 

#### Description of Task to be completed?
- Use a proper context as a response for unknown routes
- Follow the rules stated for API specification
- 'createdOn' should be an integer
- Remove image from the request body as it is an optional feature
- Follow naming convention for route.js and test.js
- Add more validations for routes
- Remove the PUT request
- Rename redflagid to id
- Add status in the body of the request
- Create a middleware for validations
- Fix code coverage issue
- Write more tests to ensure code coverage

#### How should this be manually tested?
- Check the new branch to ensure that all feedback listed in the descriptions above are reflected

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#162439563](https://www.pivotaltracker.com/n/projects/2226946)

#### Screenshots (if appropriate)
N/A

#### Questions:
- Please I need a feedback with regards my Travis and coveralls if it is now properly set up according to recommended standards

### CODE REVIEW

File: server/controllers/redFlagsController.js [hunk 1]
```diff
@@ -31,12 +31,19 @@ class RedFlagsController {
     * @memberOf redFlagsController
     */
   static getARedFlagById(req, res) {
-    const id = req.params.redFlagId;
-    const redFlag = redFlags.find(redFlagItem => +redFlagItem.redFlagId === +id);
+    const { id } = req.params;
+    const redFlag = redFlags.find(r => r.id === parseInt(req.params.id));
```
Comment (fob413) [inline server/controllers/redFlagsController.js]: is `r` a good variable name? Please correct this and any other places where you might be using non-descriptive variable names.

File: server/controllers/redFlagsController.js [hunk 2]
```diff
@@ -31,12 +31,19 @@ class RedFlagsController {
...
     if (!redFlag) {
-      return res.status(404).json({ message: `Red-flag with the given Id ${id} does not exist` });
+      return res.status(404).json({
+        error: 404,
+        message: `Red-flag with the given Id ${id} does not exist`
+      });
     }
-    return res.json({ message: 'Red-flags search was successful', redFlag });
+    return res.status(200).json({
+      message: 'Red-flags search was successful',
```
Comment (fob413) [inline server/controllers/redFlagsController.js]: Please go over the product specification document. This `message` is not supposed to be here. Please correct this and other endpoints

File: server/controllers/redFlagsController.js [hunk 3]
```diff
@@ -103,14 +106,20 @@ class RedFlagsController {
     * @memberOf redFlagsController
     */
   static updateRedFlagLocation(req, res) {
-    // eslint-disable-next-line radix
-    const redFlag = redFlags.find(r => r.redFlagId === parseInt(req.params.redFlagId));
+    const redFlag = redFlags.find(r => r.id === parseInt(req.params.id));
     if (!redFlag) {
-      return res.status(404).json({ message: 'Red-flag with the given location does not exist' });
+      return res.status(404).json({ status: 404, message: 'Red-flag with the given location was not found' });
```
Comment (fob413) [inline server/controllers/redFlagsController.js]: This line is too long, please adjust it and any other similar long lines

**Review by fob413 — CHANGES_REQUESTED** [review]

File: server/controllers/redFlagsController.js (hunk 1 above)
Comment (Raymond-Osy) [reply server/controllers/redFlagsController.js]: Noted and corrected, thanks.

File: server/controllers/redFlagsController.js (hunk 2 above)
Comment (Raymond-Osy) [reply server/controllers/redFlagsController.js]: Noted and corrected

File: server/controllers/redFlagsController.js (hunk 3 above)
Comment (Raymond-Osy) [reply server/controllers/redFlagsController.js]: Noted and corrected

File: server/controllers/redFlagsController.js (hunk 1 above)
Comment (Raymond-Osy) [reply server/controllers/redFlagsController.js]: Noted and corrected

File: server/middleware/validator.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        error: 'comments not found in the request body',
+      });
+    }
+    if (typeof (comment) !== 'string') {
+      return res.status(400).send({
+        status: 400,
+        error: 'comments must be a string',
+      });
+    }
+    if (!location || comment.trim() === '') {
```
Comment (fob413) [inline server/middleware/validator.js]: is there a reason why you're checking `comment.trim()` twice?

File: server/middleware/validator.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        error: 'location must contain a string',
+      });
+    }
+    next();
+  }
+
+  static redFlagLocationValidator(req, res, next) {
+    const { location } = req.body;
+
+    if (!location || typeof (location) !== 'string') {
```
Comment (fob413) [inline server/middleware/validator.js]: Is there a reason why you're not checking for `location.trim` here?

File: server/middleware/validator.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        error: 'location must be present and must be a string',
+      });
+    }
+    next();
+  }
+
+  static redFlagCommentValidator(req, res, next) {
+    const { comment } = req.body;
+
+    if (!comment || typeof (comment) !== 'string') {
```
Comment (fob413) [inline server/middleware/validator.js]: is there a reason for not checking `comment.trim()` here?

**Review by fob413 — CHANGES_REQUESTED** [review]
Anything you're saving from the `req.body` must be trim'ed. Please correct this

File: server/middleware/validator.js (hunk 5 above)
Comment (Raymond-Osy) [reply server/middleware/validator.js]: Noted and corrected

File: server/middleware/validator.js (hunk 4 above)
Comment (Raymond-Osy) [reply server/middleware/validator.js]: Noted and corrected

File: server/middleware/validator.js (hunk 6 above)
Comment (Raymond-Osy) [reply server/middleware/validator.js]: Noted and corrected

**Pull request merged**